src/include/processor/operator/scan/scan_node_table.h

```diff
@@ -71,9 +71,8 @@ class ScanNodeTable final : public ScanTable {
 private:
     common::vector_idx_t currentTableIdx;
     // TODO(Guodong): Refactor following three fields into a vector of structs.
-    std::vector<std::unique_ptr<ScanNodeTableInfo>> infos;
+    std::vector<ScanNodeTableInfo> nodeInfos;
```

Comment:
Aren't we following the convention of keeping info and states (local and shared) separately? Why move the local state into info?

Reply:
I'm doing this for simplicity. I'll try to make a decision once I figured out how to turn on IndexScan properly.